File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char,int> mp = {
            {'I',1},
            {'V',5},
            {'X',10},
            {'L',50},
            {'C',100},
            {'D',500},
            {'M',1000},
        };
        
        int sum=0;
        for(int i=0;i<s.size();i++)
        {
            auto it=mp.find(s[i]);
            auto It=mp.find(s[i+1]);
            if(it->second<It->second){
                sum+=(It->second-it->second);
                i++;
            }else
                sum+=it->second;
        }
        return sum;
    }
};

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character,Integer>mp=new HashMap<>();
        mp.put('I',1);
        mp.put('I',1);
        mp.put('V',5);
        mp.put('X',10);
        mp.put('L',50);
        mp.put('C',100);
        mp.put('D',500);
        mp.put('M',1000);
        int sum=0,i=0;

        for(;i<s.length()-1;i++){
            if(mp.get(s.charAt(i))<mp.get(s.charAt(i+1))){
                sum+=mp.get(s.charAt(i+1))-mp.get(s.charAt(i));
                i++;
            }else{
                sum+=mp.get(s.charAt(i));
            }
        }
        if(i==s.length()-1)return sum+mp.get(s.charAt(i));
        else return sum;
        
        
    }
}